Split Bottom Menu into Elements Tower menu and Item Stash menu

Praytic opened the pull request:
![image](https://github.com/Praytic/youtd2/assets/10060411/54867c05-428e-4a25-a606-1d81b5cf9786)

- `BottomMenuBar` has being renamed to `BottomMenu` and it was reduced to couple of buttons. Each button opens a dedicated menu to control towers/items.
- Picked items are now flying to the `ItemStashMenuButton` in the `BottomMenu`, not to a particular rarity filter. This is because rarity filters can now be hidden by the `ItemStashMenu` if it's not open and it'll be confusing if items will fly to the invisible node.
- Item filters can now be toggled independently of each other.
- Tower element filters have no modulate if they are selected or were upgrade to at least 1 lvl. In addition, the currently selected element filter is highlighted.
- `BottomMenuBar` is not in its final form, I will rework it to look better. Right now it's only used to open Towers and Items menus.

Kvel2D reviewed (changes requested):
The merge from main is messed up because changes from multiple commits are completely overwritten.
Some of the commits:
https://github.com/Praytic/youtd2/commit/22e8d96b302811a0462bf35af0aa49a68985f6be
https://github.com/Praytic/youtd2/commit/c5192a396784314501033af420ed4e3b6d7a12ae
https://github.com/Praytic/youtd2/commit/7705a4da626efecb9bd7cee8b2e658f6abafc816

There may be more.

Kvel2D commented on Scenes/HUD/BottomMenuBar.tscn:
```diff
@@ -236,17 +131,8 @@ grow_vertical = 2
 mouse_force_pass_scroll_events = false
 theme = ExtResource("1_r4ocg")
 script = ExtResource("6")
-_item_stash_menu = NodePath("HBoxContainer/CentralMenu2/VBoxContainer/MarginContainer/ScrollContainer/StashMarginContainer/ItemStashMenu")
```
Removal of all of these export vars is incorrect because they are still used in script.
Causes null errors during runtime.
For example, if you press on the "Show only regular items" button you get this error:

> Invalid call. Nonexistent function 'set_current_item_type_filter' in base 'Nil'.

Praytic replied:
I will remove them completely in the next commits.

Kvel2D commented:
How is the new tower menu supposed to be reopened once it's closed?
There's no indication anywhere on how to do it.

Praytic commented:
> How is the new tower menu supposed to be reopened once it's closed?
> There's no indication anywhere on how to do it.
Let me add a button to the BottomMenuBar which opens this menu in the next couple of commits.

Kvel2D commented on Scenes/HUD/Buttons/TowerButton.gd:
```diff
@@ -18,7 +18,10 @@ func _ready():
 	set_rarity(TowerProperties.get_rarity(_tower_id))
 	set_icon(TowerProperties.get_icon_texture(_tower_id))
-	set_tier_icon(_tower_id)
+	if Globals.game_mode == GameMode.enm.TOTALLY_RANDOM:
```
I think this is supposed to include both "totally random" and "random with upgrades". So use `Globals.game_mode_is_random()`.

Praytic replied:
No, it's supposed to use only this mode, because in "random with upgrade" we drop only towers with the first upgrade.

Kvel2D replied:
Ok, I understand.

Kvel2D commented on Scenes/HUD/WaveStatus.tscn:
```diff
@@ -4,7 +4,6 @@
 [ext_resource type="Script" path="res://Scenes/HUD/WaveStatus.gd" id="2_fsc15"]
 
 [node name="WaveStatus" type="VBoxContainer" node_paths=PackedStringArray("_label", "_stats_label", "_timer_label")]
-custom_minimum_size = Vector2(650, 0)
```
This overwrites a change from main branch.

Praytic replied:
Restored

Kvel2D reviewed (changes requested):
Can't start the game. Get the following errors:

>   Attempt to open script 'res://Scenes/HUD/Buttons/UnitButtonContainer.gd' resulted in error 'File not found'.
  Failed loading resource: res://Scenes/HUD/Buttons/UnitButtonContainer.gd. Make sure resources have been imported by opening the project in the editor at least once.
  scene/resources/resource_format_text.cpp:283 - res://Scenes/HUD/HoradricMenu2.tscn:505 - Parse Error: 
  Failed loading resource: res://Scenes/HUD/HoradricMenu2.tscn. Make sure resources have been imported by opening the project in the editor at least once.
  scene/resources/resource_format_text.cpp:283 - res://Scenes/HUD.tscn:143 - Parse Error: 
  Failed loading resource: res://Scenes/HUD.tscn. Make sure resources have been imported by opening the project in the editor at least once.
  scene/resources/resource_format_text.cpp:283 - res://Scenes/GameScene.tscn:119 - Parse Error: 
  Failed loading resource: res://Scenes/GameScene.tscn. Make sure resources have been imported by opening the project in the editor at least once.
  editor/editor_data.cpp:626 - Index p_idx = 1 is out of bounds (edited_scene.size() = 1).

Praytic commented on Scenes/HUD/ItemStashMenu.gd:
```diff
@@ -1,32 +1,49 @@
+extends PanelContainer
 
 
-# This UI element displays items which are currently in the
```
It was a mistake, reverted.

Praytic commented:
> Can't start the game. Get the following errors:
> 
> > Attempt to open script 'res://Scenes/HUD/Buttons/UnitButtonContainer.gd' resulted in error 'File not found'.
> > Failed loading resource: res://Scenes/HUD/Buttons/UnitButtonContainer.gd. Make sure resources have been imported by opening the project in the editor at least once.
> > scene/resources/resource_format_text.cpp:283 - res://Scenes/HUD/HoradricMenu2.tscn:505 - Parse Error:
> > Failed loading resource: res://Scenes/HUD/HoradricMenu2.tscn. Make sure resources have been imported by opening the project in the editor at least once.
> > scene/resources/resource_format_text.cpp:283 - res://Scenes/HUD.tscn:143 - Parse Error:
> > Failed loading resource: res://Scenes/HUD.tscn. Make sure resources have been imported by opening the project in the editor at least once.
> > scene/resources/resource_format_text.cpp:283 - res://Scenes/GameScene.tscn:119 - Parse Error:
> > Failed loading resource: res://Scenes/GameScene.tscn. Make sure resources have been imported by opening the project in the editor at least once.
> > editor/editor_data.cpp:626 - Index p_idx = 1 is out of bounds (edited_scene.size() = 1).

It works on a fresh build.

Kvel2D commented:
Found problems:

### Runtime errors
![2023-12-16 08_07_43-Window](https://github.com/Praytic/youtd2/assets/13269470/5da405a7-7474-49a1-9b38-69d271582362)

![2023-12-16 08_10_01-Window](https://github.com/Praytic/youtd2/assets/13269470/da93f8e7-9ad3-4f94-bc80-1d8cf23685e3)

### Can't move items to most containers
From tower to item stash
![2023-12-16 07_57_44-Window](https://github.com/Praytic/youtd2/assets/13269470/1de606cd-d561-4c7a-9b9a-83fa0cf2aa9d)
From tower to item stash button/area - that whole UI panel should be a valid item drop target
![2023-12-16 07_57_21-Window](https://github.com/Praytic/youtd2/assets/13269470/5cf6e51d-7cfb-4746-a4df-91cfed6960cd)
From horadric cube to item stash
![2023-12-16 08_06_14-Window](https://github.com/Praytic/youtd2/assets/13269470/56ba8479-ec9d-4512-9e47-45a73f7436ce)

### Escape doesn't close tower stash and item stash
Added an issue for this: #327
A bit complicated to implement. Should do this after this PR.

Praytic commented:
> Runtime errors

Fixed

> Can't move items to most containers

Fixed

> Escape doesn't close tower stash and item stash

Will be addressed in a different PR

> Rarity filters are next to horadric cube but they affect item stash

Swapped panels

> Can't scroll item stash and tower menu with mousewheel

This is complicated issue and will addressed in a different PR

> Item slots in tower menu got bigger and now towers with 6 slots are weird

Reduced the size
